### PR TITLE
nixos/cri-o: add aufs, devmapper, and zfs to storageDrivers 

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Description of changes
Closes #112443
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/virtualisation/cri-o.nix [hunk 1]
```diff
@@ -126,7 +126,7 @@ in
       documentation = [ "https://github.com/cri-o/cri-o" ];
       wantedBy = [ "multi-user.target" ];
       after = [ "network.target" ];
-      path = [ cfg.package ];
+      path = [ cfg.package ] ++ optional (cfg.storageDriver == "zfs") config.boot.zfs.package;
```
Comment (zowoq) [inline nixos/modules/virtualisation/cri-o.nix:129]: Please change it to how podman does this.

https://github.com/NixOS/nixpkgs/blob/4d628f4b73a6ed8dece33b40c8704d8164dd252d/nixos/modules/virtualisation/podman/default.nix#L8-L11

Comment (zowoq) [reply nixos/modules/virtualisation/cri-o.nix]: Sorry, I meant for it to be moved `crioPackage` as well.

Comment (zowoq) [reply nixos/modules/virtualisation/cri-o.nix]: @SuperSandro2000 Do you want to do the `crioPackage` change and preserve your commit signing or should I do it?

Comment (SuperSandro2000) [reply nixos/modules/virtualisation/cri-o.nix]: applied your diff

**Review by zowoq — APPROVED** [review]

**Pull request merged**